packages/flutter/lib/src/gestures/tap_and_drag.dart

```diff
@@ -1138,26 +1159,39 @@ sealed class BaseTapAndDragGestureRecognizer extends OneSequenceGestureRecognize
   @override
   String get debugDescription => 'tap_and_drag';
 
+  /// Calculates the update delta from the given pointer event.
+  OffsetPair _calculateDragUpdateDelta(PointerEvent event) {
+    final Offset localDelta = event.localDelta;
+
+    if (localDelta == Offset.zero) {
+      return OffsetPair.zero;
+    }
+
+    final Offset correctedLocalPosition = _initialPosition.local + localDelta;
+    final Matrix4? localToGlobalTransform = event.transform == null ? null : Matrix4.tryInvert(event.transform!);
+    final Offset globalUpdateDelta = PointerEvent.transformDeltaViaPositions(
+      transform: localToGlobalTransform,
+      untransformedDelta: localDelta,
+      untransformedEndPosition: correctedLocalPosition,
+    );
+    return OffsetPair(local: localDelta, global: globalUpdateDelta);
+  }
+
   void _acceptDrag(PointerEvent event) {
     if (!_wonArenaForPrimaryPointer) {
       return;
     }
     if (dragStartBehavior == DragStartBehavior.start) {
-      _initialPosition = _initialPosition + OffsetPair(global: event.delta, local: event.localDelta);
+      _initialPosition += OffsetPair(global: event.delta, local: event.localDelta);
+      _currentPosition = _initialPosition;
     }
     _checkDragStart(event);
     if (event.localDelta != Offset.zero) {
-      final Matrix4? localToGlobal = event.transform != null ? Matrix4.tryInvert(event.transform!) : null;
-      final Offset correctedLocalPosition = _initialPosition.local + event.localDelta;
-      final Offset globalUpdateDelta = PointerEvent.transformDeltaViaPositions(
-        untransformedEndPosition: correctedLocalPosition,
-        untransformedDelta: event.localDelta,
-        transform: localToGlobal,
-      );
-      final OffsetPair updateDelta = OffsetPair(local: event.localDelta, global: globalUpdateDelta);
-      _correctedPosition = _initialPosition + updateDelta; // Only adds delta for down behaviour
-      _checkDragUpdate(event);
-      _correctedPosition = null;
+      if (_dragState == _DragState.accepted) {
```

Comment:
I think this case was originally only meant for cases when `DragStartBehavior.down` is set, see: https://github.com/flutter/flutter/blob/dd437c7a604ba2180ff6fba40f52613593af4f04/packages/flutter/lib/src/gestures/monodrag.dart#L775-L795 . When the behavior is set to `DragStartBehavior.down` we send a `start` event with the initial position from the down event and a `update` event with the delta since the down event vs when the behavior is set to `DragStartBehavior.start` we only send the `start` event with the initial position offset by the delta since the down event and no `update` event.

Reply:
Added the assertion.